src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -34,6 +34,9 @@
 
     from .base import Candidate, Requirement
 
+    C = TypeVar("C")
+    Cache = Dict[Link, C]
```

Comment:
This seems unrelated but I'm going to assume it helps prepare for some part of your next PR. I'm not bothered enough to suggest that it be moved to that PR, though.